### PR TITLE
Switch possible parts of post_fail_hooks to serial terminal

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#120735](https://progress.opensuse.org/issues/120735)
- Follow up to: de0279c (#15977) and 4bad8a0 (#16058)
- Verification run: [extra_tests_textmode@64bit](https://pdostal-server.suse.cz/tests/3516), [extra_tests_gnome@64bit](https://pdostal-server.suse.cz/tests/3515)

### CODE REVIEW

**Comment by foursixnine** [comment]
@pdostal, do you have some VR's for us? TW/Leap15.5 and JeOS? :)

**Comment by pdostal** [comment]
@foursixnine done.

**Review by okurz — CHANGES_REQUESTED** [review]
The commit message could have a bit more details, e.g. what issue is fixed, why this needs to change, to which versions of which product it applies, link to a bug or a fate entry, the choices you made, etc. Also see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages.

**Review by okurz — COMMENTED** [review]
Please describe in the git commit message what the benefit of the change is, at best make it prominent in the git commit message subject line. To me it's not obvious that the advantage is

File: lib/Utils/Logging.pm [hunk 1]
```diff
@@ -183,8 +184,7 @@ This method will call several other log gathering methods from this class.
 =cut
 
 sub export_logs {
-    select_log_console();
-    save_screenshot();
```
Comment (okurz) [inline lib/Utils/Logging.pm:187]: Someone put in these `save_screenshot` calls in on purpose. I am not sure it's a good idea to remove those. I suggest you show in more details in the git commit message how the output looked in before and how it's better now (if it actually is)

File: lib/consoletest.pm [hunk 2]
```diff
@@ -51,13 +50,6 @@ sub post_fail_hook {
     return if get_var('NOLOGS');
     record_avc_selinux_alerts();
     $self->SUPER::post_fail_hook;
-    # at this point the instance is shutdown
-    return if (is_public_cloud() || is_openstack());
-    select_console('log-console');
-    remount_tmp_if_ro;
```
Comment (okurz) [inline lib/consoletest.pm:57]: Where did you move this code? Or do you just remove it without replacement?

File: lib/Utils/Logging.pm (hunk 1 above)
Comment (pdostal) [reply lib/Utils/Logging.pm:187]: As I switched from log console to serial terminal, do we still really need this?

File: lib/consoletest.pm (hunk 2 above)
Comment (pdostal) [reply lib/consoletest.pm:57]: This is already executed in `Utils::Logging::export_logs()` which is called from `opensusebasetest::post_fail_hook()` and so it makes no sense to be duplicated also here.

File: lib/Utils/Logging.pm (hunk 1 above)
Comment (okurz) [reply lib/Utils/Logging.pm:187]: Well, I don't know that so you need to convince us :)

File: lib/consoletest.pm (hunk 2 above)
Comment (okurz) [reply lib/consoletest.pm:57]: Ok, cool

File: lib/Utils/Logging.pm (hunk 1 above)
Comment (pdostal) [reply lib/Utils/Logging.pm:187]: This does not make sense. After my change we don't switch to log console so we do not have to take picture of it.

What I'm saying is that before we switched to log console, took screenshot of it and then do something with it. But now, we don't switch to it at all.

Comment (okurz) [reply lib/Utils/Logging.pm:187]: oh, yes. I see that now. Forgive me, I read it the other way around where a screenshot was taken from wherever we would come frome *before* we select the log console. So all is good.

**Review by okurz — APPROVED** [review]

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

**Comment by DimStar77** [comment]
Responsible for https://openqa.opensuse.org/tests/2968643#step/kate/4

Firefox is the 'real' failure, but then we end up on serial console, reload a VM snapshot state of LASTGOOD (which is on X-console), continue typing over VNC until we end up in kate does that ensure_installed, which interacts on serial and is completely consfused

**Pull request merged**